gems/sorbet-runtime/lib/types/props/private/deserializer_generator.rb

```diff
@@ -57,15 +57,11 @@ def self.generate(props, defaults)
               begin
                 #{transformation}
               rescue NoMethodError => e
-                T::Configuration.soft_assert_handler(
-                  'Deserialization error (probably unexpected stored type)',
-                  storytime: {
-                    klass: self.class,
-                    prop: #{prop.inspect},
-                    value: val,
-                    error: e.message,
-                    notify: 'djudd'
-                  }
+                self.class.decorator.raise_deserialization_error(
```

Comment:
Should this really live on the `decorator`? Wouldn't that mean that we're expecting to allow people to customize this?

I think that if we put the `raise_deserialization_error` on `DeserializerGenerator` we can call it like

```
T::Props::Private::DeserializerGenerator.raise_deserialization_error
```

and not have to worry about indirections. (tbh, i have no real idea what `decorator` is, and even though it's probably not possible anymore, i kind of still wish for us to remove it at some point, so i try to find ways to avoid mentioning it in new code).